packages/dd-trace/test/llmobs/sdk/integration.spec.js

```diff
@@ -212,8 +217,81 @@ describe('end to end sdk integration tests', () => {
       }
     ]
 
-    check(exptected, evaluationMetrics)
+    check(expected, evaluationMetrics)
 
     Date.now.restore()
   })
+
+  describe('distributed', () => {
+    it('injects and extracts the proper llmobs context', () => {
+      payloadGenerator = function () {
+        const carrier = {}
+        llmobs.trace({ kind: 'workflow', name: 'parent' }, workflow => {
+          tracer.inject(workflow, 'text_map', carrier)
+        })
+
+        const spanContext = tracer.extract('text_map', carrier)
+        tracer.trace('new-service-root', { childOf: spanContext }, () => {
+          llmobs.trace({ kind: 'workflow', name: 'child' }, () => {})
+        })
+      }
+
+      const { llmobsSpans } = run(payloadGenerator)
+      expect(llmobsSpans).to.have.lengthOf(2)
+
+      expect(getTag(llmobsSpans[0], 'ml_app')).to.equal('test')
+      expect(getTag(llmobsSpans[1], 'ml_app')).to.equal('test')
+    })
+
+    it('injects the local mlApp', () => {
+      payloadGenerator = function () {
+        const carrier = {}
+        llmobs.trace({ kind: 'workflow', name: 'parent', mlApp: 'span-level-ml-app' }, workflow => {
+          tracer.inject(workflow, 'text_map', carrier)
+        })
+
+        const spanContext = tracer.extract('text_map', carrier)
+        tracer.trace('new-service-root', { childOf: spanContext }, () => {
+          llmobs.trace({ kind: 'workflow', name: 'child' }, () => {})
+        })
+      }
+
+      const { llmobsSpans } = run(payloadGenerator)
+      expect(llmobsSpans).to.have.lengthOf(2)
+
+      expect(getTag(llmobsSpans[0], 'ml_app')).to.equal('span-level-ml-app')
+      expect(getTag(llmobsSpans[1], 'ml_app')).to.equal('span-level-ml-app')
+    })
+
+    it('injects a distributed mlApp', () => {
+      payloadGenerator = function () {
+        let carrier = {}
+        llmobs.trace({ kind: 'workflow', name: 'parent' }, workflow => {
+          tracer.inject(workflow, 'text_map', carrier)
+        })
+
+        // distributed call to service 2
+        let spanContext = tracer.extract('text_map', carrier)
+        carrier = {}
+        tracer.trace('new-service-root', { childOf: spanContext }, () => {
+          llmobs.trace({ kind: 'workflow', name: 'child-1' }, child => {
+            tracer.inject(child, 'text_map', carrier)
+          })
+        })
+
+        // distributed call to service 3
+        spanContext = tracer.extract('text_map', carrier)
+        tracer.trace('new-service-root', { childOf: spanContext }, () => {
+          llmobs.trace({ kind: 'workflow', name: 'child-2' }, () => {})
+        })
+      }
+
+      const { llmobsSpans } = run(payloadGenerator)
+      expect(llmobsSpans).to.have.lengthOf(3)
+
+      expect(getTag(llmobsSpans[0], 'ml_app')).to.equal('test')
+      expect(getTag(llmobsSpans[1], 'ml_app')).to.equal('test')
+      expect(getTag(llmobsSpans[2], 'ml_app')).to.equal('test')
```

Comment:
:+1: nice. It would also be worth having a case that asserts the precedence between the span vs distributed vs global too but I'll leave it up to you for if you want to do this